lua/terrortown/migrations/shared/2024_01_22_10_26_sampleMigration.lua

```diff
@@ -0,0 +1,19 @@
+--- @ignore
+-- Sample Migration file for server
+-- Shows the use of migration commands
+local states = {endMessage = "This is the end of all revert migration changes."}
+local function Upgrade(cmd)
+	print("You successfully upgraded to a system with migrations.")
+end
+
+local function Downgrade(cmd)
+	print(cmd.endMessage)
+end
+
+migrations.Add(
+	migrations.CreateCommand(
+		states,
+		Upgrade,
+		Downgrade
+	)
+)
```

Comment:
Id rather imagine an auto load of the folder and something we do already eg with hudelements and just have some global table or just return the up and down methods from the file.

Also i dont get what this states thingy is for.

Reply:
Yeah sounds good, just didnt want to spend any time for it until we are sure about the overall workings.

States is derived from the [command pattern](https://en.wikipedia.org/wiki/Command_pattern) and shall make clear that everything that command needs is inside the class. This would contain the before and after states, that can be saved for example. This makes a command absolutely standalone in programming languages were you have to manually address memory for example or the variable contents change.

Generally used for forward and revert mechanics in photoshop or similar.

Reply:
This is overly complicated to me and i don't see any issue that is mitigated by using this. 
Also the migration functions don't need any state. I'd prefer if we keep it simple here.

Really i'd envision this to replicate the Laravel system, as it is quite simple and easy.